src/platform/log/zephyr/log.c

```diff
@@ -40,3 +82,76 @@ mender_log_exit(void) {
     /* Nothing to do */
     return MENDER_OK;
 }
+
+#ifdef CONFIG_MENDER_DEPLOYMENT_LOGS
+mender_err_t
+mender_deployment_logs_activate(void) {
+    log_backend_activate(&log_backend_mender_depl_logs, NULL);
+    return MENDER_OK;
+}
+
+mender_err_t
+mender_deployment_logs_deactivate(void) {
+    MENDER_NDEBUG_UNUSED mender_err_t ret;
+
+    /* In case there is an incomplete log message, save it. */
+    if (0 != msg_tmp_idx) {
+        msg_tmp[msg_tmp_idx++] = '\0';
+        ret                    = mender_storage_deployment_log_append((const char *)msg_tmp, msg_tmp_idx);
+        assert(MENDER_OK == ret);
+        msg_tmp_idx = 0;
+    }
+
+    log_backend_deactivate(&log_backend_mender_depl_logs);
+    return MENDER_OK;
+}
+
+static int
+depl_logs_log(uint8_t *data, size_t length, MENDER_ARG_UNUSED void *ctx) {
+    MENDER_NDEBUG_UNUSED mender_err_t ret;
+
+    /* This function can be called even for every character of a log message logged. */
+    for (uint8_t i = 0; i < length; i++) {
+        if ('\r' == (char)data[i]) {
+            /* throw away carriage-returns */
+            continue;
+        } else if ('\n' == (char)data[i]) {
+            /* newline terminates a log message */
+            msg_tmp[msg_tmp_idx++] = '\0';
+            ret                    = mender_storage_deployment_log_append((const char *)msg_tmp, msg_tmp_idx);
+            assert(MENDER_OK == ret);
+            msg_tmp_idx = 0;
+        } else {
+            /* just append the character to the message being constructed */
+            msg_tmp[msg_tmp_idx++] = data[i];
+        }
+    }
+
+    /* This function ought to return the number of bytes processed so even in
+       case of a failure we have no other choice. */
+    return length;
+}
+
+static void
+panic(struct log_backend const *const backend) {
+    log_backend_deactivate(backend);
+}
+
+static void
+dropped(MENDER_ARG_UNUSED const struct log_backend *const backend, uint32_t cnt) {
+    log_output_dropped_process(&depl_log_output, cnt);
+}
+
+static void
+process(MENDER_ARG_UNUSED const struct log_backend *const backend, union log_msg_generic *msg) {
+    /* We only care about warning and error log messages */
+    if (msg->log.hdr.desc.level > LOG_LEVEL_WRN) {
+        return;
+    }
+
+    uint32_t          flags           = log_backend_std_get_flags() & ~LOG_OUTPUT_FLAG_COLORS;
+    log_format_func_t log_format_func = log_format_func_t_get(LOG_OUTPUT_TEXT);
+
+    log_format_func(&depl_log_output, &msg->log, flags);
+}
+#endif /* CONFIG_MENDER_DEPLOYMENT_LOGS */
```

Comment:
All in all the deployment logs code is around 75% of the file now, and there is nothing shared with the rest of the "normal" logging. Consider moving it to `log_deployment.c`.

Reply:
I did consider that. But since there was (and would be again) basically nothing in `log.c`, I don't think we need that in a separate file. Perhaps in the future there's more to add to this file and the split will make more sense. But I'm leaving it as is for now.